packages/coil-extension/src/popup/components/AnimateTippingOpacityWrapper.tsx

```diff
@@ -0,0 +1,72 @@
+// * NOTE: All AnimateWrapper components can be removed and should not effect layout
+
+import React from 'react'
+import { motion } from 'framer-motion'
+
+import { useRouter } from '../context/routerContext'
+import { ROUTES } from '../constants'
+
+//
+// Component
+//
+
+/*
+    This animation is only for primary navigation transitions. 
+    This does not handle the sub route navigation transitions such as from TipView -> TipConfirmView -> TipCompleteView
+*/
+export const AnimateTippingOpacityWrapper = (
+  props: React.PropsWithChildren<any>
+) => {
+  const { path, previousPath } = useRouter()
+  const transitionDuration = 0.1
+
+  console.log('- current: ', path)
+  console.log('-prev: ', previousPath)
+  // not sure where to put this, using it to determine the direction
+  const isSubRoute =
+    path.includes('tipping') && previousPath.includes('tipping')
+
+  const noAnimation = path === ROUTES.tippingComplete
+
+  const primaryVariants = {
+    initial: {
+      opacity: 0
+    },
+    enter: {
+      opacity: 1,
+      transition: {
+        duration: transitionDuration
+      }
+    },
+    exit: {
+      opacity: 0,
+      transition: {
+        duration: noAnimation ? 0 : transitionDuration
+      }
+    }
+  }
+
+  // staticVariants is set to simply eliminate the animation if the navigation is for a primary route
```

Comment:
<img width="395" alt="image" src="https://user-images.githubusercontent.com/525211/156953447-9d94c4dd-d8ed-4573-a612-56bc8c1f2e40.png">

Lot's of copy/paste here

Reply:
Are you referring to the static variants?

Reply:
A lot of the blocks of code look quite similar, with the same comments

Reply:
Yeah those are the animation settings for each animation wrapper. I specifically don't share those because they are not all the same and it makes them easier to manage and read like this.

Reply:
I'm not religiously dry, but it's 4 very similar blocks
Actually for the non acquainted reader, it's probably easier to read some parameterized function, so you can easily see what is different between them.

Reply:
Think of decoupled > dry in these cases. Decoupling specific components makes them more managable and easier to figure out whats going on vs creating a complexity to remove similar code. I did remove the use of another variant where possible though since I can simply pass in the animation config directly and I also renamed the variables to be a bit more clear.